### ISSUE TITLE
Segfault on UNKNOWN_REQUEST_TYPE

### ISSUE THREAD

**Issue by rosik** [tarantool/tarantool#6451]
**Bug description**

A change in cartridge revealed a segfault.

The change: https://github.com/tarantool/cartridge/commit/b3a5e45f5fa850e8e003f187d1349a86350f6f78

The backtrace:

```log
main-2 | Segmentation fault
main-2 |   code: SEGV_MAPERR
main-2 |   addr: 0x8
main-2 |   context: 0x7f455b0ff640
main-2 |   siginfo: 0x7f455b0ff770
main-2 |   rax      0x0                0
main-2 |   rbx      0x7f4561400be0     139935961058272
main-2 |   rcx      0x5                5
main-2 |   rdx      0x5575332f5e50     93961858276944
main-2 |   rsi      0x7f455b0ffc60     139935857245280
main-2 |   rdi      0x0                0
main-2 |   rsp      0x7f455b0ffcd0     139935857245392
main-2 |   rbp      0x7f455b0ffcd0     139935857245392
main-2 |   r8       0x5575332ec2d0     93961858237136
main-2 |   r9       0x5575332ec2a0     93961858237088
main-2 |   r10      0x5575332a8e30     93961857961520
main-2 |   r11      0x5575332ec2f0     93961858237168
main-2 |   r12      0x7f455b0ffcf0     139935857245424
main-2 |   r13      0x7f455b0ffd28     139935857245480
main-2 |   r14      0x7f4561400a90     139935961057936
main-2 |   r15      0x5575315b4f40     93961827602240
main-2 |   rip      0x5575315ac39f     93961827566495
main-2 |   eflags   0x10202            66050
main-2 |   cs       0x33               51
main-2 |   gs       0x0                0
main-2 |   fs       0x0                0
main-2 |   cr2      0x8                8
main-2 |   err      0x4                4
main-2 |   oldmask  0x0                0
main-2 |   trapno   0xe                14
main-2 | Current time: 1631886549
main-2 | Please file a bug at http://github.com/tarantool/tarantool/issues
main-2 | Attempting backtrace... Note: since the server has already crashed,
main-2 | this may fail as well
main-2 | #0  0x5575315e13cd in print_backtrace+d
main-2 | #1  0x5575314f6489 in _ZL12sig_fatal_cbiP9siginfo_tPv+d9
main-2 | #2  0x7f456409b3c0 in funlockfile+60
main-2 | #3  0x5575315ac39f in diag_raise+f
main-2 | #4  0x5575315ae1aa in _Z23applier_resume_to_stateP7applier13applier_stated+17a
main-2 | #5  0x55753159bebb in _ZL21bootstrap_from_masterP7replica+2b
main-2 | #6  0x5575315a08ab in box_cfg+67b
main-2 | #7  0x5575314f6799 in load_cfg+239
main-2 | #8  0x5575315b4f56 in _ZL13lbox_cfg_loadP9lua_State+16
main-2 | #9  0x5575315f93f7 in lj_BC_FUNCC+34
main-2 | #10 0x5575315fd7b8 in lua_pcall+78
main-2 | #11 0x5575315caec8 in luaT_call+18
main-2 | #12 0x5575315c5c59 in lua_fiber_run_f+59
main-2 | #13 0x5575314f58f1 in _ZL16fiber_cxx_invokePFiP13__va_list_tagES0_+11
main-2 | #14 0x5575315ddb30 in fiber_loop+30
main-2 | #15 0x5575317b9e64 in coro_init+44
```

In the test, there're two cartridge instances. One serves runs remote-control server (lua iproto implementation) and another one tries to bootstrap.

Before the patch, remote-control didn't reply to unknown requests but closed the socket instead. And after the patch it started to reply fairly with `box.error.UNKNOWN_REQUEST_TYPE` ([code](https://github.com/tarantool/cartridge/blob/b3a5e45f5fa850e8e003f187d1349a86350f6f78/cartridge/remote-control.lua#L117-L130)).

The second server connects to the first one and sends `IPROTO_VOTE = 68` request. But for some reason, it can't handle the error response and crashes.

Place `tarantool --version` output here.

```console
$ tarantool -v
Tarantool 1.10.11-2-ge30b44140
Target: Linux-x86_64-RelWithDebInfo
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/opt/tarantool-install -DENABLE_BACKTRACE=ON
Compiler: /usr/bin/cc /usr/bin/c++
C_FLAGS: -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-gnu-alignof-expression -fno-gnu89-inline -Wno-cast-function-type
CXX_FLAGS: -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type
```

**Comment by sergepetrenko** [comment]
@rosik, I can't reproduce the error. Tried on both current master, 1.10.14 and 1.10.11-2

I made the master reply with UNKNOWN_REQUEST_TYPE to IPROTO_VOTE, but nothing crashed.
Replica ignores the error (like it's intended) and continues joining. Master's logs have
```
2022-09-26 14:20:08.494 [59648] iproto iproto.cc:1282 E> ER_UNKNOWN_REQUEST_TYPE: Unknown request type 68
```

Also looks like your code encodes the error the same way Tarantool does. I don't see what could go wrong there.

Need some more info on how to reproduce the issue from your side.
